typescript/sdk/src/utils/MultiGeneric.ts

```diff
@@ -1,25 +1,58 @@
+import { AllChains } from '../consts/chains';
 import { ChainMap, ChainName, Remotes } from '../types';
 
 export class MultiGeneric<Chain extends ChainName, Value> {
-  constructor(protected readonly chainMap: ChainMap<Chain, Value>) {}
+  constructor(public readonly chainMap: ChainMap<Chain, Value>) {}
 
-  protected get(chain: Chain) {
-    return this.chainMap[chain];
+  /**
+   * Get value for a chain
+   * @throws if chain is invalid or has not been set
+   */
+  protected get(chain: Chain): Value {
+    if (!chain || !AllChains.includes(chain)) {
+      throw new Error(`Invalid chain ${chain}`);
+    }
+    const value = this.chainMap[chain] ?? null;
+    if (!value) {
+      throw new Error(`No chain value found for ${chain}`);
+    }
+    return value;
+  }
+
+  /**
+   * Get value for a chain
+   * @returns value or null if chain value has not been set
+   */
+  protected tryGet(chain: Chain): Value | null {
+    if (!chain || !AllChains.includes(chain)) {
+      return null;
+    }
+    return this.chainMap[chain] ?? null;
   }
 
-  protected set(chain: Chain, value: Value) {
+  /**
+   * Set value for a chain
+   * @throws if chain is invalid or has not been set
+   */
+  protected set(chain: Chain, value: Value): Value {
+    if (this.tryGet(chain)) {
+      throw new Error(`Value already exists for chain ${chain}`);
+    }
     this.chainMap[chain] = value;
+    return value;
   }
 
-  chains = () => Object.keys(this.chainMap) as Chain[];
+  chains(): Chain[] {
+    return Object.keys(this.chainMap) as Chain[];
+  }
 
-  apply(fn: (n: Chain, dc: Value) => void) {
+  apply(fn: (n: Chain, dc: Value) => void): void {
```

Comment:
I am to blame for this bad name and function but we need to rename this lol

Reply:
Will rename to `forEach`